codingTest_jye/비밀지도.js

```diff
@@ -0,0 +1,37 @@
+function solution(n, arr1, arr2) {
+  //배열 선언방법 잘 기억해두자. 빈 배열을 n개 만들어주었다.
+  let answer = new Array(n).fill([]);
+  //이진수로 바꾸기
+  for (let i = 0; i < n; i++) {
+    arr1[i] = arr1[i].toString(2);
+    arr1[i] = arr1[i].split("");
+    arr2[i] = arr2[i].toString(2);
+    arr2[i] = arr2[i].split("");
+  }
+
+  //비교를 위해 문자열의 길이를 같게 만들어준다
+  for (let i = 0; i < n; i++) {
+    if (arr1[i].length !== n) {
+      while (arr1[i].length < n) {
+        arr1[i].unshift("0");
+      }
+    }
+    if (arr2[i].length !== n) {
+      while (arr2[i].length < n) {
+        arr2[i].unshift("0");
+      }
+    }
+  }
```

Comment:
문자열의 길이를 같게 만들어주는 방법!

**padStart()** 메서드 쓰시면 긴 코드가 한 줄로 줄어드는 마법을 볼 수 있습니다.
https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

검색 키워드는 **'js 문자열 길이 맞추기'**

```suggestion javascript
for (let i = 0; i < n; i++) {
    arr1[i] = arr1[i].toString(2).padStart(n, 0);
    arr2[i] = arr2[i].toString(2).padStart(n, 0);
}
```

Reply:
레전드 padStart 이해해씁니다...